metricflow/query/query_parser.py

```diff
@@ -643,59 +679,90 @@ def _parse_metric_names(
                     metric_references.extend(list(input_metrics))
         return tuple(metric_references)
 
-    def _parse_linkable_element_names(
+    def _parse_linkable_elements(
         self,
-        qualified_linkable_names: Sequence[str],
         metric_references: Sequence[MetricReference],
+        qualified_linkable_names: Optional[Sequence[str]] = None,
+        linkable_elements: Optional[Sequence[QueryParameter]] = None,
     ) -> QueryTimeLinkableSpecSet:
         """Convert the linkable spec names into the respective specification objects."""
-        qualified_linkable_names = [x.lower() for x in qualified_linkable_names]
+        assert not (qualified_linkable_names and linkable_elements)
+
+        structured_names: List[StructuredLinkableSpecName] = []
+        if qualified_linkable_names:
+            qualified_linkable_names = [x.lower() for x in qualified_linkable_names]
+            structured_names = [StructuredLinkableSpecName.from_name(name) for name in qualified_linkable_names]
+        elif linkable_elements:
+            for linkable_element in linkable_elements:
+                parsed_name = StructuredLinkableSpecName.from_name(linkable_element.name)
+                if parsed_name.time_granularity:
+                    raise ValueError(
+                        "Time granularity must be passed in the `grain` attribute for `group_by` query param."
```

Comment:
Will this always be a group_by? I think at some point it might be a filter or order by or something, but I'm honestly not sure. @DevonFulcher, do you know?

Reply:
Yes, you will soon be able to specify `Dimension('metric_time').grain('WEEK')` in the where param or filter field.

Reply:
I can update this to say `Time granularity must be passed in the `grain` attribute of the {type({linkable_element})} parameter`, but I'm not sure the type we receive here will match the type of the parameter they're seeing on the client side 🤔

Reply:
Updating this to say `Must use object syntax for `grain` parameter if `date_part` is requested.` We will probably want to change that error message when we stop supporting dundered granularity syntax.